app/services/common_platform/api/search_prosecution_case.rb

```diff
@@ -11,10 +11,51 @@ def initialize(params)
 
       def call
         check_response_status
-
+        check_for_blank_defendants(prosecution_cases)
         record_prosecution_cases
       end
 
+      def check_for_blank_defendants(prosecution_cases)
+        if !prosecution_cases.nil? && !prosecution_cases.empty?
+          prosecution_cases.each do |prosecution_case|
+            if !prosecution_case["defendantSummary"].nil? && !prosecution_case["defendantSummary"].empty?
+              prosecution_case["defendantSummary"].each do |defendant|
+                # If the defendant is missing these values in their summary
+                if (!defendant["defendantFirstName"]) && (!defendant["defendantLastName"]) && (!defendant["defendantDOB"]) && (!defendant["defendantNINO"])
+                  if @empty_defendants.nil?
+                    @empty_defendants = []
+                  end
+                  # Add this defendant to an array of empty defendants (we need to also verify they don't have a hearing summary further down)
+                  @empty_defendants.push(defendant["defendantId"])
+                end
+              end
+            end
+
+            # We need to check the hearing summaries for the blank defendants
+            if !prosecution_case["hearingSummary"].nil? && !prosecution_case["hearingSummary"].empty?
```

Comment:
Each of these blocks of code could be its own method too.

Reply:
Basically we want the code readable enough that we don't need the comments.

Reply:
I've split into a single method to check if the array is empty, as there are multiple checks here for the same sort of thing. Hopefully it is more readable than before (and reduces some duplication).